runtime/devnet/src/config/api.rs

```diff
@@ -1,14 +1,23 @@
 use crate::{config::assets::TrustBackedAssetsInstance, fungibles, Runtime, RuntimeCall};
 use codec::{Decode, Encode, MaxEncodedLen};
 use frame_support::traits::Contains;
+use pop_chain_extension::ReadStateHandler;
 
 /// A query of runtime state.
 #[derive(Encode, Decode, Debug, MaxEncodedLen)]
 #[repr(u8)]
-pub enum RuntimeRead<T: fungibles::Config> {
+pub enum RuntimeRead {
 	/// Fungible token queries.
 	#[codec(index = 150)]
-	Fungibles(fungibles::Read<T>),
+	Fungibles(fungibles::Read<Runtime>),
+}
+
+impl ReadStateHandler<Runtime> for RuntimeRead {
```

Comment:
Implementing this trait for the enum seems weird. It is the runtime that is deciding to match an enum variant to a pallet function, so it might feel more natural to implement the trait for the Runtime or for a local Extension struct.

An example that comes to mind is the XcmExecutor struct which is used with pallet-xcm. If we consider the Extension as a subsystem of pallet-contracts, would modelling it in a similar way make it easier. That is, we provide a generic Extension which the runtime wraps in its own Extension struct. Any requirements are then implemented for the Extension struct rather than the current Config trait you have implemented for the Runtime.